### DISCUSSION TITLE
can not mutate nested object in store

### DISCUSSION THREAD

**Discussion by llz383455526** [pmndrs/valtio#665] [Q&A]
I had used valtio in web applications, it works fine! 
But in React Native App, I encountered a problem。

In react native project, when i mutate object in store, show error:
``` TypeError: target property is non-configurable and non-writable, and set trap value differs from target property value ```

``` typescript
// store/NewsCreate.ts

const initNews = {
  replyedTimeline: undefined,   // type is undefined | {ts:'', user: {name: '', age: ''}}
};
export const newsStore = proxy(initNews);
```

```typescript
// screen/index.ts

import { newsStore } from '@/store/NewsCreate';

...
const news = useSnapshot(newsStore);
useEffect(() => {
    if (route.params?.newsItem ) {
      newsStore.replyedTimeline = route.params.newsItem;  // problem happened here
    }
  }, [route.params?.newsItem]);
```

**Comment by dai-shi** [comment]
Thanks for reporting.

Can you add `console.log(Object.getOwnPropertyDescriptors(route.params.newItem))` before the error and share what it prints?

**Reply by llz383455526** [reply]
Thanks for reply. 

![image](https://user-images.githubusercontent.com/788738/218621934-1e319904-6e3b-4b47-a9a0-976e9158698d.png)

the property of newsItem is not writable, because i passed a  snapshot value to the route params.

Now i passed store value to route params directly, it works fine

**Reply by dai-shi** [reply]
Oh, so the `route.params` are actually valtio snapshots.
Yeah, store value would be a solution.

Can you try v1.10.2? If the problem persists, we can probably create a small reproduction.

**Reply by afkcodes** [reply]
i am also facing the same issue
![Screenshot 2023-07-21 at 1 21 17 AM](https://github.com/pmndrs/valtio/assets/17390257/bc6ec471-2ad4-4312-8962-7107a0e48a0a)

**Reply by mbaharan** [reply]
Downgrading to version 1.10.7 solved my problem.

**Reply by dai-shi** [reply]
@mbaharan Is your issue the same as #764 ?

I think it's different from this #665.

Ah, do you mean v1.10.7 works for you, and v1.11.0 fails?

**Reply by afkcodes** [reply]
interesting thing is when i added valtio to my project 3 days back it was working fine the version i installed was 1.10.7 but suddenly yesterday started throwing the error. the same code that used to work, i did not update the versions though

**Reply by mbaharan** [reply]
@dai-shi I started receiving the same error: `TypeError: target property is non-configurable and non-writable, and set trap value differs from target property value`. Just downgrading to the previous version solved my problem. I just also checked to see if the variable is configurable and writable and it was. Not sure why I was getting that error. The problem started with version 1.11.0.

**Reply by dai-shi** [reply]
@mbaharan Are you using RN? Please see #764 and check if your issue is the same.
